File: JavaScript/997. Find the Town Judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
    if (trust.length == 0 && n == 1){
        return 1;
    }
    let trusted=new Array(n+1).fill(null);
    for(var i = 0; i<trust.length; i++){
        trusted[trust[i][0]]--;
        trusted[trust[i][1]]++;
    }
    for(var i = 0; i<trusted.length;i++){
        if(trusted[i]==n-1){return i;}
    }
    return -1;
};

File: C++/997. Find the Town Judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        if (trust.size() == 0 && n == 1){
            return 1;
        }
        vector<int> trusted(n+1);
        for(int i = 0; i<trust.size(); i++){
            trusted[trust[i][0]]--;
            trusted[trust[i][1]]++;
        }
        for(int i = 0; i<trusted.size();i++){
            if(trusted[i]==n-1){return i;}
        }
        return -1;
    }
};
